surface/include/pcl/surface/impl/convex_hull.hpp

```diff
@@ -214,15 +214,16 @@ pcl::ConvexHull<PointInT>::performReconstruction2D (PointCloud &hull, std::vecto
   }
 
   int num_vertices = qh num_vertices;
-  hull.resize (num_vertices);
-  memset (&hull.points[0], hull.size (), sizeof (PointInT));
+
+  std::size_t hull_size = hull.size();
+  hull.clear();
+  hull.resize(hull_size);
 
   vertexT * vertex;
   int i = 0;
 
   std::vector<std::pair<int, Eigen::Vector4f>, Eigen::aligned_allocator<std::pair<int, Eigen::Vector4f> > > idx_points (num_vertices);
   idx_points.resize (hull.size ());
```

Comment:
This is quite confusing. We allocate num_vertices, and resize to hull.size

Is there any relation between these 2 numbers (one is always >, < or == another)?

Reply:
I walked through the relevant sources codes `surface/include/pcl/surface/impl/convex_hull.hpp` `surface/include/pcl/surface/convex_hull.h` `test/surface/test_convex_hull.cpp` and had better understanding about the algorithms and function calls. 

I think, in 2D convex hull reconstruction, the `hull.size()` is exactly the same with `num_vertices` after qhull finds all vertices of convex hull successfully. And it also equals to `idx_points.size()` (which used as a temporary variable for sorting the vertices), and `polygons[0].vertices.size()` (which only use 1 element in 2D reconstruction).

Reply:
And I was wondering if I could changes the code here to:
https://github.com/PointCloudLibrary/pcl/blob/87f1d83377abbbc29b48b268960cdb96e02d327b/surface/include/pcl/surface/impl/convex_hull.hpp#L216-L221
```c++
int num_vertices = qh num_vertices;
hull.clear();
hull.resize(num_vertices);
```
Thus, `hull.size()` would get the same value as `num_vertices`, and be used in the next context.

Reply:
Not sure if the `hull.clear()` is necessary, but else it looks good

Reply:
Perhaps `hull` was used and stored other data inside before passing to this function? I'm not sure about this.😆

Reply:
Thanks for the investigation. In that case,
```suggestion
  AlignedVector<std::pair<int, Eigen::Vector4f>> idx_points (num_vertices);
```
where the following can go in `common/pcl/types.h` (or not. I just find it to be too verbose)
```cpp
template <T>
AlignedVector = std::vector<T, Eigen::aligned_allocator<T>>;
```

Reply:
Using defined data type makes the code pretty clean! Thank you!